tests/test_api.py

```diff
@@ -2026,6 +2029,155 @@ def test_operations(self):
+
+        with self.assertRaisesRegex(InvalidValueError, GROUP_NAME_MISSING):
+           api.add_group(self.ctx, "Example", None, None)
+
+        transactions = Transaction.objects.filter(created_at__gt=trx_date)
+        self.assertEqual(len(transactions), 0)
+
+    def test_group_name_is_empty(self):
+        """Check if it fails when group name is an empty string"""
+
+        trx_date = datetime_utcnow()
+
+        with self.assertRaisesRegex(InvalidValueError, GROUP_NAME_MISSING):
+           api.add_group(self.ctx, "Example", None, "")
+
+        transactions = Transaction.objects.filter(created_at__gt=trx_date)
+        self.assertEqual(len(transactions), 0)
+
+    def test_organization_not_found(self):
+        """Check if group is created if organization is not found"""
+
+        trx_date = datetime_utcnow()
+
+        with self.assertRaisesRegex(NotFoundError, ORGANIZATION_NOT_FOUND_ERROR.format(name='Exampe')):
+              api.add_group(self.ctx, "Exampe", None, "subgroup")
+
+        transactions = Transaction.objects.filter(created_at__gt=trx_date)
+        self.assertEqual(len(transactions), 0)
+
+    def test_find_parent(self):
+        """ Check if parent is able to be found"""
+        api.add_group(self.ctx, "Example", None, "parent")
+        child = api.add_group(self.ctx, "Example", "parent" , "child")
+        self.assertIsInstance(child, Group)
+        self.assertEqual(child.name, "child")
+        self.assertEqual(child.organization, self.org)
+
+    def test_parent_not_found(self):
+        """ Check if a group cannot be created when parent is not found"""
+
+        api.add_group(self.ctx, "Example", None, "suborg")
+
+        with self.assertRaisesRegex(NotFoundError, NOT_FOUND_ERROR.format(entity="parent")):
+            api.add_group(self.ctx, "Example", "parent", "child")
+
+
+class TestDeleteGroup(TestCase):
+    """Unit tests for delete_group"""
+
+    def setUp(self):
+        """Load initial dataset"""
+
+        self.user = get_user_model().objects.create(username='test')
+        self.ctx = SortingHatContext(self.user)
+        self.org = api.add_organization(self.ctx, name='Example')
+
+    def test_delete_group(self):
+        """Check if group can be deleted"""
+        api.add_group(self.ctx, "Example", None, "suborg")
+        group = api.delete_group(self.ctx, self.org.name,"suborg")
+        self.assertIsInstance(group, Group)
+        self.assertEqual(group.name, "suborg")
+        self.assertEqual(group.organization, self.org)
+
+    def test_organization_is_none(self):
+        """Check if it fails when organization name is `None`"""
+
+        api.add_group(self.ctx, "Example", None, "suborg")
+        trx_date = datetime_utcnow()
+        with self.assertRaisesRegex(InvalidValueError, GROUP_ORG_NAME_MISSING):
+            api.delete_group(self.ctx, None, "suborg")
+        transactions = Transaction.objects.filter(created_at__gt=trx_date)
+        self.assertEqual(len(transactions), 0)
```

Comment:
Please, organize this test (and the other ones) so we can read them better. For example:

```
api.add_group(self.ctx, "Example", None, "suborg")

trx_date = datetime_utcnow()

with self.assertRaisesRegex(InvalidValueError, GROUP_ORG_NAME_MISSING):
     api.delete_group(self.ctx, None, "suborg")

transactions = Transaction.objects.filter(created_at__gt=trx_date)
self.assertEqual(len(transactions), 0)
```

Reply:
Done 👍